src/core/update_initial_conditions.jl

```diff
@@ -173,7 +175,7 @@ function _make_initial_conditions!(
         ini_conds = Vector{InitialCondition}(undef, length_devices)
         set_initial_conditions!(ic_container, key, ini_conds)
         for (ix, dev) in enumerate(devices)
-            val_ = get_val_func(dev, key)
+            val_ = get_val_func(dev, key, device_formulation, variable_type)
```

Comment:
I don't fully understand why we are passing here a variable_type

Reply:
To use the get_variable_initial_value I need variable type, device and formulation.